client/app/components/packages/pas-form/edit.hbs

```diff
@@ -697,22 +697,29 @@
           <button
             type="button"
             class="button expanded large"
-            disabled={{not this.isSaveable}}
-            {{on "click" this.save}}
+            disabled={{or (not this.isSaveable) this.isSavingOrSubmitting}}
+            {{on "click" (perform this.save)}}
             data-test-save-button
           >
             {{!-- TODO: Make this button work with the action passed in @save--}}
-            Save
+            {{#if this.save.isRunning}}
+              <FaIcon @icon="spinner" @spin="true" size="5x" class="light-gray" />
+            {{else}}
+              Save
+            {{/if}}
           </button>
           <button
             type="button"
             class="button expanded large secondary"
-            disabled={{not this.isSubmittable}}
+            disabled={{or (not this.isSubmittable) this.isSavingOrSubmitting}}
```

Comment:
Do you think this would be better up in the component class just so that we're not keeping button disabled logic in two places ( the template and the class file)?

Reply:
It should fit fine in `this.isSubmittable` (and same for the `isSaveable` getter). Because it's true that _if it's in the process of saving, it isn't in a saveable/submittable state_ until that task has finished.

Reply:
I think I understand what you mean now! See comment above